packages/expo/src/hooks/useSSO.ts

```diff
@@ -0,0 +1,95 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { EnterpriseSSOStrategy, OAuthStrategy, SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+
+import { errorThrower } from '../utils/errors';
+
+export type StartSSOFlowParams = {
+  unsafeMetadata?: SignUpUnsafeMetadata;
+} & (
+  | {
+      strategy: OAuthStrategy;
+    }
+  | {
+      strategy: EnterpriseSSOStrategy;
+      identifier: string;
+    }
+);
+
+export type StartSSOFlowReturnType = {
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+};
+
+export function useSSO() {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startSSOFlow(startSSOFlowParams: StartSSOFlowParams): Promise<StartSSOFlowReturnType> {
+    if (!isSignInLoaded || !isSignUpLoaded) {
```

Comment:
❓  Should we throw an error here when the `strategy` is not provided?

Reply:
I think the types would already caught it, unless it's a current pattern for us to include runtime validations for pure JS usage on Expo

Reply:
Yeah I mentioned it mostly to be a bit more strict and also for developer that not using Typescript would be convenient to see an error and fix it, definitely not a blocker though!